lib/features/auth/widgets/terms_and_conditions_dialog.dart

```diff
@@ -1,81 +1,94 @@
 import 'package:flutter/material.dart';
 import 'package:givt_app/app/injection/injection.dart';
 import 'package:givt_app/core/enums/enums.dart';
-import 'package:givt_app/core/network/country_iso_info.dart';
+import 'package:givt_app/features/email_signup/cubit/email_signup_cubit.dart';
 import 'package:givt_app/features/family/shared/design/components/components.dart';
 import 'package:givt_app/features/family/shared/widgets/texts/shared_texts.dart';
 import 'package:givt_app/features/family/utils/utils.dart';
 import 'package:givt_app/l10n/l10n.dart';
 import 'package:go_router/go_router.dart';
 
-class TermsAndConditionsDialog extends StatelessWidget {
+class TermsAndConditionsDialog extends StatefulWidget {
   const TermsAndConditionsDialog({
     required this.typeOfTerms,
     super.key,
   });
   //todo refactor this to use the title and the instead of the [TypeOfTerms]
   final TypeOfTerms typeOfTerms;
 
+  @override
+  State<TermsAndConditionsDialog> createState() =>
+      _TermsAndConditionsDialogState();
+}
+
+class _TermsAndConditionsDialogState extends State<TermsAndConditionsDialog> {
+  final _cubit = getIt<EmailSignupCubit>();
+  late String countryIso;
+  @override
+  void initState() {
+    super.initState();
+    _cubit.init();
+    _getCountry();
```

Comment:
I would use "then" here because you can't await the async function in the init and you do want to get the country before the widget initially builds considering you are using a late variable. Else you might get a not initialized error if the sharedpreferences take too long.

Reply:
Ignore, i am sleepy, then is not synchronous so it doesn't matter

Reply:
So I would either already set the country to the same value initially as the cubit (NL) so that even if it builds earlier you don't get an error, or I would reintroduce the futurebuilder